src/internal/m365/onedrive/permission.go

```diff
@@ -82,6 +83,56 @@ func getCollectionMetadata(
 	return meta, nil
 }
 
+// Unlike permissions, link shares inherit them from all the parents
+// of an item and not just the direct parent.
+func computePreviousLinkShares(
+	ctx context.Context,
+	originDir path.Path,
+	parentMetas map[string]metadata.Metadata,
+) ([]metadata.LinkShare, error) {
+	var (
+		parent     path.Path
+		linkShares []metadata.LinkShare
+	)
+
+	parent = originDir
+
+	for {
+		newParent, err := parent.Dir()
+		if err != nil {
+			return nil, clues.New("getting parent").WithClues(ctx)
+		}
+
+		if parent == newParent {
+			break
+		}
+
+		parent = newParent
+
+		ictx := clues.Add(ctx, "parent_dir", parent)
+
+		drivePath, err := path.ToDrivePath(parent)
+		if err != nil {
+			return nil, clues.New("transforming dir to drivePath").WithClues(ictx)
+		}
+
+		if len(drivePath.Folders) == 0 {
+			break
+		}
+
+		meta, ok := parentMetas[parent.String()]
+		if !ok {
+			return nil, clues.New("no metadata found for parent folder: " + parent.String()).WithClues(ictx)
+		}
+
+		if meta.SharingMode == metadata.SharingModeCustom {
```

Comment:
does adding a link to an item automatically set it to custom sharing mode? (this should probably have a comment saying so too so others have that context)

Reply:
Any change in permissions will change the mode to 'custom'. Let me add a note though.

> `custom` is something we use to represent on our end based on the `shared` value in the item response in delta.